Raster transparency API improvements + cleanups

nyalldawson opened the pull request:
- Make classes less like to be created with uninitialized members
- Use opacity instead of transparency, to match API from elsewhere in QGIS
- Add equality operators and Python repr
- Add explicit setting for includeMinimum/Maximum to QgsRasterTransparency.TransparentSingleValuePixel. Allows (via
API only) control over whether the extremities of the range should be included in the transparency.
- Add more unit tests for these classes

No user facing changes here!

nyalldawson committed:
Cleanup QgsRasterTransparency API

- Make classes less like to be created with uninitialized members
- Use opacity instead of transparency, to match API from elsewhere in
  QGIS

nyalldawson committed:
[api] Add explicit setting for includeMinimum/Maximum

to QgsRasterTransparency.TransparentSingleValuePixel. Allows (via
API only) control over whether the extremities of the range
should be included in the transparency.

Also add equality operator, repr to these classes, and additional
tests

rouault commented on src/core/raster/qgsrastertransparency.cpp:
```diff
@@ -97,7 +87,7 @@ int QgsRasterTransparency::alphaValue( double value, int globalTransparency ) co
   if ( myTransparentPixelFound )
   {
-    return static_cast< int >( static_cast< float >( globalTransparency ) * ( 1.0 - ( myTransparentPixel.percentTransparent / 100.0 ) ) );
+    return static_cast< int >( static_cast< float >( globalTransparency ) * myTransparentPixel.opacity );
```
a transparency multiplied by an opacity... That gives me some headaches :-)  Is globalTransparency properly named or is it a globalOpacity ?

nyalldawson replied:
It's actually an alpha value :rofl: (0-255, integer). Let me fix this too!

nyalldawson replied:
Ok, should be fixed by [729c814](https://github.com/qgis/QGIS/pull/56598/commits/729c81431151efd1e241a3e253a548eaa29fb454)

rouault commented on src/core/raster/qgsrastertransparency.cpp:
```diff
@@ -133,7 +123,7 @@ int QgsRasterTransparency::alphaValue( double redValue, double greenValue, doubl
   if ( myTransparentPixelFound )
   {
-    return static_cast< int >( static_cast< float >( globalTransparency ) * ( 1.0 - ( myTransparentPixel.percentTransparent / 100.0 ) ) );
+    return static_cast< int >( static_cast< float >( globalTransparency ) * myTransparentPixel.opacity );
```
same remark as above

nyalldawson committed:
Further cleanups to QgsRasterTransparency API

Avoid confusing mix of transparency vs opacity, multiple scaling
and round trips through integers.

nyalldawson committed:
Use vector instead of list for better performance

This container is expected to have only a handful of items in it
in most cases and will rarely change => QVector is a better choice

nyalldawson committed:
Update test mask
